### DISCUSSION TITLE
[Email] Unexpected error while sending an email:  Sendmail exited with code 1

### DISCUSSION THREAD

**Discussion by gremo** [directus/directus#5281] [Archive]
Using docker image v9-rc61, I can't send email. I've double checked my configuration and it seems fine to me:

`EMAIL_FROM`: no-reply@xxxxxxxxxxx
`EMAIL_TRANSPORT`: sendmail
`EMAIL_SENDMAIL_NEW_LINE`: unix
`EMAIL_SENDMAIL_PATH`: /usr/sbin/sendmail

On my server:

```bash
-bash-4.2$ which sendmail
/usr/sbin/sendmail
```

When I try to invite a new user, the mailer doesn't work. From the docker output:

> 14:20:55 ✨ request completed POST 400 /users/invite 63ms
14:21:01 ⚠️  [Email] Unexpected error while sending an email:
14:21:01 ⚠️  Sendmail exited with code 1
14:21:01 ✨ request completed POST 200 /users/invite 110ms

Any clue?

**Comment by rijkvanzanten** [comment]
There's an error thrown by `sendmail` itself "Sendmail exited with code 1", so you'll have to look for the answer outside of Directus I'm afraid.. 

Are you able to send emails directly through `sendmail` on your server?

**Reply by gremo** [reply]
Why closing the issue so early? Yes, sendmail works correctly on the server. 

Maybe it's a PATH issue with docker build?

`PATH` environment variable for the container: `/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin`

**Reply by rijkvanzanten** [reply]
I moved the issue to a Q&A discussion, as it's seemingly a configuration problem and not a bug in directus.

When you say "works on the server", did you do anything to make it available in the docker container? I'm not sure if the docker image ships with a copy of sendmail included ( @WoLfulus? ).

**Reply by gremo** [reply]
I think that `PATH` is used to expose executables from the host machine to the container, but I'm not sure. Any help is much appreciated, thanks 👍 .

This is a screenshot from the Plesk Docker extension, Any other env variable is working just fine (like `DB_USER`, `USER_INVITE_URL_ALLOW_LIST`). Also volume mapping is working fine.

`PATH` actually contains` /usr/sbin/` which is exactly where `sendmail` is (on host).

![Cattura](https://user-images.githubusercontent.com/1532616/116415107-6372c400-a839-11eb-8c21-9f661cac5ae2.PNG)

**Reply by rijkvanzanten** [reply]
PATH is just the environment variable that's used in the docker container. That being said, the Docker container doesn't share a filesystem with your actual server, so that path `/usr/local/sbin` etc might exist on your server, but won't inside the docker container 👍🏻 

I'd highly recommend using SMTP for email connections if you're in this sort of containerized environment to avoid conflicts with different filesystems etc 🙂

**Reply by gremo** [reply]
Oh, now I understand @rijkvanzanten thank you! I'm going to install Directus directly on the server without Docker. I'm pretty sure `sendmail` will work.

**Reply by sapkra** [reply]
@rijkvanzanten I have the same problem. For me it seems more like a bug of the directus docker image. Probably it doesn't include the sendmail command, so the default config is broken.

Edit: I checked it and sendmail is installed but it doesn't work anyway.

Edit 2: It seems like the problem is that sendmail is trying to connect to an smtp server by no one is running inside the container. When trying to send an email manually I get the following error: `sendmail: Cannot open mail:25`.

[This answer on StackOverflow](https://stackoverflow.com/a/45498504/3382901) was quite helpful.

**Reply by rijkvanzanten** [reply]
@WoLfulus?
